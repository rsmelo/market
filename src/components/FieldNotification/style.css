.fieldNotification {
  position: absolute;
  top: 100%;
  line-height: 120%;
  font-size: 0.8em;
  left: 0;
  padding: 0;
}

.error {
  color: var(--color-error);
}

.success {
  color: var(--color-success);
}

