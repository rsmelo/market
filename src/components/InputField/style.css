.container {
  position: relative;
  width: 100%;
}

.input {
  border-radius: 0;
  border: 1px solid #e4e4e4;
  box-shadow: none;
  transition: border-color 0.2s;
  padding: 4px 8px;
  width: 100%;
}

.input:focus {
  border-color: #999999;
  box-shadow: none;
  outline: 0;
}
