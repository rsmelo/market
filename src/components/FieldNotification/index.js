import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import style from './style.css'

const FieldNotification = ({ message, type }) => (
  <span className={classNames(style.fieldNotification, style[type])}>
    {message}
  </span>
)

FieldNotification.propTypes = {
  message: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['success', 'error']).isRequired,
}

export default FieldNotification
