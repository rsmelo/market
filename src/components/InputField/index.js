import React from 'react'
import PropTypes from 'prop-types'

import FieldNotification from '../FieldNotification/'
import style from './style.css'

const InputField = ({
  input,
  type,
  id,
  placeholder,
  meta: { touched, error },
  ...props
}) => (
  <div className={style.input}>
    <input {...input} type={type} id={id} placeholder={placeholder} {...props} />
    {touched &&
      (error && <FieldNotification type="error" >{error}</FieldNotification>)}
  </div>
)

InputField.defaultProps = {
  placeholder: null,
  maxLength: null,
}

InputField.propTypes = {
  input: PropTypes.shape({}).isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool.isRequired,
    error: PropTypes.string,
  }).isRequired,
  placeholder: PropTypes.string,
  type: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  maxLength: PropTypes.number,
}

export default InputField
