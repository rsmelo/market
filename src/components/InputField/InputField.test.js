import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import InputField from './index'
import FieldNotification from '../FieldNotification'

describe('InputField', () => {
  it('should render correctly', () => {
    const props = {
      input: {},
      type: 'text',
      id: 'name',
      meta: {
        touched: true,
        error: 'teste',
      },
    }
    const tree = renderer
      .create(<InputField {...props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should show error if touched and has error', () => {
    const props = {
      input: {},
      type: 'text',
      id: 'name',
      meta: {
        touched: true,
        error: 'teste',
      },
    }
    const component = shallow(<InputField {...props} />)
    const notification = component.find(FieldNotification)
    expect(notification).toHaveLength(1)
  })
  it('should not show error if was nottouched and has error', () => {
    const props = {
      input: {},
      type: 'text',
      id: 'name',
      meta: {
        touched: false,
        error: 'teste',
      },
    }
    const component = shallow(<InputField {...props} />)
    const notification = component.find(FieldNotification)
    expect(notification).toHaveLength(0)
  })
})
