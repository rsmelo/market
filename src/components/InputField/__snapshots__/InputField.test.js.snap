// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should render correctly 1`] = `
<div
  className={undefined}
>
  <input
    id="name"
    maxLength={null}
    placeholder={null}
    type="text"
  />
  <span
    className=""
  >
    teste
  </span>
</div>
`;
